C-Plus-Plus/cp/weird algorithm.cpp

```diff
@@ -1,15 +1,41 @@
+/*****
+
+Consider an algorithm that takes as input a positive integer n. If n is even, the algorithm divides it by two, and if n is odd, the algorithm multiplies it by three and adds one. The algorithm repeats this, until n is one. For example, the sequence for n=3 is as follows:
+3→10→5→16→8→4→2→1
+
+Your task is to simulate the execution of the algorithm for a given value of n.
+
+Input
+
+The only input line contains an integer n.
+
+Output
+
+Print a line that contains all values of n during the algorithm.
+
+Example :
+Input:
+3
+
+Output:
+3 10 5 16 8 4 2 1
+
+*****/
+
+
 #include<iostream>
 using namespace std;
  
 int main(){
     long long n;
     cin>>n;
-    while(true){
-        cout << n << " ";
-        if (n == 1) break;
-        if (n%2 == 0) n /= 2;
+    while(true)
+    {
+        cout << n << " ";         
+        if (n == 1) break;       // breaking here coz finally we have to reach here
```

Comment:
Great work. Just a small amendment. You need to add comments above the lines of code and not alongside. 
Also, mention the complexity of the solution. Rest, everything is picture perfect!